### PR TITLE
Added a new package @types/required

### PR DESCRIPTION
**Pull request by monolithed**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

The current PR is a proposal to add a new type [Required](https://github.com/Microsoft/TypeScript/issues/15012)

### CODE REVIEW

File: types/required/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts"
```
Comment (andy-ms) [inline types/required/tsconfig.json]: Forgot `"required-tests.ts"`

**Comment by andy-ms** [comment]
DefinitelyTyped isn't the place for this.

**Comment by monolithed** [comment]
Why not?

**Comment by andy-ms** [comment]
DefinitelyTyped is for describing types of existing packages. You appear to be authoring something entirely new.

**Comment by monolithed** [comment]
The package name "required" is already used. I can create a new package like "type-required" and publish one in the npm registry. Is it OK for you?

**Comment by andy-ms** [comment]
Well, I can't stop you! :man_shrugging:

**Comment by andy-ms** [comment]
Closing to prevent accidental merge.

**Pull request closed without merging**